core/imageroot/var/lib/nethserver/node/actions/get-facts/50get

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (C) 2022 Nethesis S.r.l.
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import json
+import sys
+import os
+
+platform = {}
+with open('/etc/os-release', encoding='utf-8') as file:
+    for line in file:
+        key, value = line.rstrip().split('=')
+        platform[key] = value.strip('"')
+
+core_version = ""
+with open('/etc/nethserver/core.env', encoding='utf-8') as file:
+    for line in file:
+        if 'CORE_IMAGE' in line:
+            print(core_version)
+            core_version = line[line.find(':') + 1:].strip()
+            break
+
+json.dump(
+    {
+        os.environ['AGENT_ID'].strip('node/'): {
```

Comment:
Little note on this, I'm assuming that AGENT_ID will change when the identifier of the node is NODE_UUID, is that the case?

Reply:
I don't want to change how the AGENT_ID is built. What I don't grasp is how to obtain NODE_UUID. Use /etc/machine-id? Generate a random UUID? ...time will tell.

BTW [.strip()](/usr/libexec/nut-driver-enumerator.sh --get-device-for-service %i) is now what you expect. You might want .removeprefix() instead.

Reply:
Well, come to think of it, we can access the `AGENT_ID/environment` on redis and read the NODE_ID from there. We can switch to UUID when the time comes. (assuming the UUID will be written there)